### PR TITLE
confluent-platform: update 5.3.0 -> 5.5.0

### PR DESCRIPTION
**Pull request by DianaOlympos**
##### Motivation for this change

The conluent cli is not open source anymore and now distributed as a binary.

Updated versions and build so that it build again.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/confluent-platform/default.nix [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-      cd confluent-cli
-      make install
-    )
-
     mkdir -p $out
+
     cp -R bin etc share src $out
     rm -rf $out/bin/windows
 
+    cp -R $confluentCli confluent-cli
```
Comment (SuperSandro2000) [inline pkgs/servers/confluent-platform/default.nix:31]: This would be a great usecase for install to also make sure the permissions are correct.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 91231` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>confluent-platform</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 91231` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>confluent-platform</li>
  </ul>
</details>

File: pkgs/servers/confluent-platform/default.nix [hunk 2]
```diff
@@ -47,15 +47,16 @@ stdenv.mkDerivation rec {
       wrapProgram $p \
         --set JAVA_HOME "${jre}" \
         --set KAFKA_LOG_DIR "/tmp/apache-kafka-logs" \
+        --set CONFLUENT_HOME "$out" \
         --prefix PATH : "${jre}/bin:${bash}/bin:${gnused}/bin"
     done
   '';
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/servers/confluent-platform/default.nix:55]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by DianaOlympos** [comment]
Closing this as the whole thing moved, i will reopen with a more recent version Soon TM

**Pull request closed without merging**